compiler-rt/test/builtins/Unit/extendhfxf2_test.c

```diff
@@ -0,0 +1,73 @@
+// RUN: %clang_builtins %s %librt -o %t && %run %t
+// REQUIRES: librt_has_extendhfxf2
+
+#include <limits.h>
+#include <math.h> // for isnan, isinf
+#include <stdio.h>
+
+#include "int_lib.h"
+
+#if HAS_80_BIT_LONG_DOUBLE && defined(COMPILER_RT_HAS_FLOAT16)
+
+long double __extendhfxf2(_Float16 f);
+
+int test_extendhfxf2(_Float16 a, long double expected) {
+  long double x = __extendhfxf2(a);
+  __uint16_t *b = (void *)&a;
```

Comment:
uint16_t

Reply:
This is defined in /usr/include/sys/int_types.h and might change depending on library. I think I'll use unsigned short directly.

Reply:
We depend on a valid <stdint.h> providing uint16_t.

Reply:
Actually, looking at this code it would be better to just use the helpers from fp_test.h: 
`uint16_t a_rep = toRep16(a)`